cli/internal/cmd/apply.go

```diff
@@ -858,3 +859,14 @@ type imageFetcher interface {
 		image, region string, useMarketplaceImage bool,
 	) (string, error)
 }
+
+type debugFileLogger struct {
+	fileHandler file.Handler
+	log         debugLog
+}
+
+func (l debugFileLogger) Debug(msg string, args ...any) {
+	l.log.Debug(msg, args...)
+
+	_ = l.fileHandler.Write(constants.InitDebugLogFile, []byte(msg+"\n"), file.OptAppend)
+}
```

Comment:
Have you considered using something like [slogmulti](https://github.com/samber/slog-multi?tab=readme-ov-file#broadcast-slogmultifanout) to handle log distribution to multiple sinks?
With the current approach, we lose a decent amount of information, e.g. no timestamps or additional fields added to the logger.

Threw something together that seems to work fine:
```Go
func newDebugFileLogger(cmd *cobra.Command, fileHandler file.Handler) (debugLog, error) {
	logLvl := slog.LevelInfo
	debugLog, err := cmd.Flags().GetBool("debug")
	if err != nil {
		return nil, err
	}
	if debugLog {
		logLvl = slog.LevelDebug
	}

	fileWriter := &fileWriter{
		fileHandler: fileHandler,
	}
	return slog.New(
		slogmulti.Fanout(
			slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{AddSource: true, Level: logLvl}),           // first handler: stderr at log level
			slog.NewJSONHandler(fileWriter, &slog.HandlerOptions{AddSource: true, Level: slog.LevelDebug}), // second handler: debug JSON log to file
		),
	), nil
}

type fileWriter struct {
	fileHandler file.Handler
}

// Write satisfies the io.Writer interface by writing a message to file.
func (l *fileWriter) Write(msg []byte) (int, error) {
	err := l.fileHandler.Write("file", msg, file.OptAppend)
	return len(msg), err
}
```

This writes any log message into a file in JSON format.
Alternatively, we could use a regular text handler to have message parity with the CLI debug output.

Reply:
I don't really think that this issue is worth bringing in a new dependency. However, I could maybe write a new Handler for those cases myself if you want?

Reply:
I don't think bringing in a new dependency for this is an issue, especially since the library itself has very few dependencies (only four, three of which we already depend on).
I also don't think putting in extra effort to re-implement and maintain similar code is worth it, when the library already does exactly what we want in an easy to use way.

Reply:
Okay, I've added your code and tested it :)